src/NotificationStack.vala

```diff
@@ -49,6 +49,10 @@ public class Gala.NotificationStack : Object {
     }
 
     public void show_notification (Meta.WindowActor notification, bool animate) {
+        if (notifications.contains (notification)) {
+            return;
+        }
```

Comment:
not really a review, but i believe you could instead do:

```vala
  public void show_notification (Meta.WindowActor notification, bool animate) requires (!notifications.contains (notification)) {
```

and then take a look in the logs to see how often this is triggered, or if it's happens when the issue occurs.